File: C++/21.状态模式/StartState.cpp
/**
 * @file StartState.cpp
 * @author {Layne} ({shu_huanglei@163.com})
 * @brief
 * @version 0.1
 * @date 2019-11-22
 *
 * @copyright Copyright (c) 2019
 *
 */

#include "StartState.h"

#include <iostream>

#include "Context.h"

StartState::StartState() {}
StartState::~StartState() {}
void StartState::DoAction(std::shared_ptr<Context> context) {
    std::cout << "Player is in start state" << std::endl;
    context->SetState(this);
}
std::string StartState::ToString() { return "Start State"; }

File: Objective-C/21.StatePattern/StartState.m
//
//  StartState.m
//  21.StatePattern
//
//  Created by Layne on 2020/11/19.
//

#import "StartState.h"

@implementation StartState

- (void)doAction:(nonnull Context *)context {
    NSLog(@"Player is in start state");
    [context setState:self];
}

- (NSString *)description {
    return @"Start State";
}

@end
